File: info/1. Алгоритмы/2. Сортировка/8. Сортировка вставками/code sample/insertion_sort.py

list_1 = [5, 0, -2, 7, 3]

for i in range(1, len(list_1)):
    paste_element = list_1[i]
    while i > 0 and list_1[i-1] > paste_element:
        list_1[i] = list_1[i-1]
        i = i-1
    list_1[i] = paste_element


print(list_1)

File: info/1. Алгоритмы/2. Сортировка/8. Сортировка вставками/code sample/InsertionSort.java

import java.util.Arrays;

public class InsertionSort {

    public static void main(String[] args) {

        int[] array = new int[] { 5, 0, -2, 7, 3 };

        for (int i = 0; i < array.length; i++) {
            int pasteElement = array[i];
            int j;
            for (j = i; j > 0; j--) {
                if (array[j - 1] <= pasteElement) {
                    break;
                }
                array[j] = array[j - 1];
            }
            array[j] = pasteElement;
        }

        System.out.println(Arrays.toString(array));
    }

}
